Assets/Scripts/ItemPickup.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ItemPickup : MonoBehaviour
+{
+    [SerializeField] private PickupSystem _pickupSystem;
+    [field: SerializeField] public ItemType ItemType {get; private set;}
+
+    private void OnTriggerEnter(Collider collision)
+    {
+        if(collision.gameObject != _pickupSystem.playerBody) return;
+        _pickupSystem.AddItem(this);
+        Destroy(gameObject);

```

Comment:
Stel je voor, het added van de item gaat mis. Dan wordt de pickup alsnog verwijderd. Je zou een bool kunnen laten returen om te checken of het echt gelukt is.